### PR TITLE
Robofac Tactical Modular Poncho

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Content "Add Hub 01 exterior armor / storage system"

#### Purpose of change
part of #53455 this part touches Hub 01 Enviro Suit (and the activity suit).

#### Describe the solution
First I think the Activity Suit just shouldn't really have storage, the scientist wearing it were wearing 5 point harnesses which would be awkward to accessorize gear around on the torso. So if they are wearing storage it would be drop leg pouches or a backpack, something that doesn't need to be it's own consolidated rig/harness.

So the solution for the activity suit is: it will copy from the robofac suit still but there wont be an additional new item for them, _at least for now_ 😈

For the Hub 01 stuff after a bit of discussion with @John-Candlebury what I've come up with is the following:

First Hub 01 will have a new outer/strapped layer item that will be a military style rip-stop poncho reinforced with a small amount of Kevlar. Similar to a thick version of this: 
image

with a bit of inspiration from the frontier with classic belted ponchos/tunics:
image

Features:
- Water Proof
- Mildly Okay defense
- Very unencumbering
-  Two waterproof pouches in the style of dry bags https://www.amazon.com/HEETA-Waterproof-Lightweight-Backpack-Transparent/dp/B07PLZP5LN/ref=sr_1_7?keywords=waterproof+bag&qid=1640791034&sr=8-7
- Two additional quick access pockets

In game:
image

As it says in the above item however **that's not all**. Hub 01 is also working on custom zombie slaying armor to work with their "Modular Defense System"

This custom armor comes in 4 pieces:

1. vambraces that cover the forearms and elbows worn as outer layer armor
2. greaves that cover the shins and knees worn as outer layer armor
3. a mantle that straps directly to the **Hub 01 Modular Defense System**. This covers the upper torso, shoulders and partially the upper arms, and lower torso. It has a shape something like this (not design just shape): 
image
4. a segmented armored skirt that straps directly to the **Hub 01 Modular Defense System**. This covers the hips and partially the upper legs. It has a shape something like this (not design just shape):
image
 
These pieces together are equivalent to a full suit of body armor but could also be worn separately. 

The pieces also come in multiple flavors. When you first meet Hub 01 their manufacturing capabilities will be limited as will their field data. As you progress and help Hub 01 they will develop better and better versions of the above armor pieces. I've categorized them into 3 Tiers of equipment.

- Tier 1 Basic Prototype - This is the initial 3d printed designs they would be testing. They would be plastic with a non hardened metal core.
image

- Tier 2 Specialized Protection - This tier consists of now refined armor design. The plastic is combined with a specialized material and the soft metal has been replaced with hardened steel core. The three variants are:
Kinetic using rubber
image

Turnout using nomex
image

Ballistic using Kevlar
image

- Tier 3 Perfected - This tier consists of the perfected armor design and also substantial advancements in fabrication tools for the faction. Using advanced military materials like reinforced aramid and UHMWPE for maximum protection while still being very lightweight.
image

Here is the data for the materials at a glance:

Name | Bash | Cut | Ballistic | Acid | Fire
-- | -- | -- | -- | -- | --
Prototype | 12 | 20 | 16 | 3.88 | .62
Kinetic | 36 | 20 | 14 | 3.25 | 1
Turnout | 20 | 20 | 14 | 3.25 | 5.5
Ballistic | 20 | 28 | 30 | 3.25 | .75
Soldier | 32 | 40 | 38 | 3.38 | 1.88

The greaves and vambraces have 75% of the values above

This is all organized into a new Hub 01 armor file called hub_armor.json the pieces are also easy to write since they build from abstracts. 

I wont put all of the items here because that would be another 15 things but an example of each is:

Spawns and locations:
The Dead Scientist in mission 1 has the modular defense system on with two basic prototype pieces
image

When you return after mission 1 you can ask about it (d)
image
image

With 10 intelligence the intercom will give you the additional prototype pieces you are missing (arms and legs)
image

Without you don't get the additional pieces
image

After that the Hub sells mid level armor pieces to you from theirs stores
image

The final hub mission has been rearranged slightly:
image

There is now an additional template stored in a box.
You can bring it to the Hub Intercom for an additional dialogue (d) (I've fixed the typo with you're since grabbing these images):
image
image

You can then ask about it and (for future proofing) other prototypes that are being worked on:
image
image
image

Currently after a day of waiting you can come back to the completed armor project. 
image
image

Getting all the pieces 
image

This UI can be extended for other high end complex science projects in the future ( I have a few in mind )

Hub 01 Mercs have on the system and ballistic pieces instead of a chest harness (lookin p. cool even without custom sprites)
image

**To Do**

- [x]  Add the Skirt, Vanbrace and Greave variants
- [x]  Add the basic versions to the game
- [x]  ~Work on quest or progression for the other tiers~
- [x]  Add missing vitals protection
- [x]  Move mantles to normal (even tho it is entirely cosmetic to do this)

#### Describe alternatives you've considered
Do something WAY less involved

#### Testing
Items are in game rn, this is still WIP tho

### CODE REVIEW

**Comment by anoobindisguise** [comment]
Out of curiosity, what's your thought process for their coverage values? Secondly, should these be OUTER and not normal-layer?

**Comment by bombasticSlacks** [comment]
Normal layer for the ablative armor doesn't matter, but it should be OUTER for consistency thanks for catching that.

Coverage is pretty simple (keep in mind the mantles don't fully cover the arms and legs they are worn with additional armor):
Mantle:
95 on upper torso and shoulders only lacking coverage around the armpits
85 on upper arms the under arms would be partially unprotected
60 on lower torso the sides and towards the waist would be open so that you could still bend freely.

Skirt: 
100 on hips since the segmented plates are consolidated
85 on upper legs since the plates might be spread leading to openings

Vambraces and Shinguards:
fully enclosed protective pieces so 95% on lower limb and joint.

**Comment by John-Candlebury** [comment]
I really like these, very creative!

Also shouldn't the mantles also offer some Vitals protection? Although I'm not very up to date on how armors are being balanced.

**Comment by bombasticSlacks** [comment]
hey I responded in discord but I didn't want to talk over people so I'll re post it here:

they should have vitals coverage that's an oversight,

Rest of this is just my opinion values are definitely up for debate: 
As far as coverage the poncho + mantle are outer not normal (the mantles being normal layer is also a mistake I'm gonna fix) so they would be worn over the nomad, survivor or hub 01 enviro suit.

So its actually competing against stuff like full plate and the ballistic vests

Ballistic vest is:
6 encumbrance 100% coverage 9 bash, 12 cut, 24 ballistic
With all the ablative plates it additionally gives:
6 encumbrance 82% coverage 25 bash, 50 cut, 50 ballistic but the plates can just shatter on you

The poncho is:
2-3 encumbrance 100% coverage 3 bash, 6 cut, 12 ballistic
With a mantle additionally it gives:
5 encumbrance 81% coverage with whatever the mantle gives for defense

so it's less encumbering than the ballistic vest 7-8 vs 12, for less defense but at the soldier mantle level that's barely the case.

Also numbers aside my rationale was it just doesn't make sense to heavily armor the stomach on a poncho your sides and stomach would need to still be able to flex as you move.

**Comment by John-Candlebury** [comment]
Seems fine balance wise too then!

Thanks for your work with these.

File: data/json/items/armor/hub_armor.json [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        "covers": [ "leg_l", "leg_r" ],
+        "specifically_covers": [ "leg_upper_l", "leg_upper_r" ]
+      }
+    ]
+  },
+  {
+    "abstract": "robofac_vambraces",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "abstract armored vambraces",
```
Comment (BrettDong) [inline data/json/items/armor/hub_armor.json:165]:
```suggestion
    "name": { "str_sp": "abstract armored vambraces" },
```

File: data/json/items/armor/hub_armor.json [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        "covers": [ "arm_l", "arm_r" ],
+        "specifically_covers": [ "arm_elbow_l", "arm_elbow_r", "arm_lower_l", "arm_lower_r" ]
+      }
+    ]
+  },
+  {
+    "abstract": "robofac_shinguards",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "abstract armored shin guards",
```
Comment (BrettDong) [inline data/json/items/armor/hub_armor.json:189]: Same here.

**Review by BrettDong — COMMENTED** [review]
Some items have incorrect plural names.

File: data/json/items/armor/hub_armor.json (hunk 1 above)
Comment (bombasticSlacks) [reply data/json/items/armor/hub_armor.json:165]: Sorry I didn't think abstract names needed plurals I'll fix this tomorrow. 

Also I appreciate you fixing all my spelling and formatting.

**Comment by bombasticSlacks** [comment]
The items in this PR are effected by #53881 and are counting as having DOUBLE protection values in the UI so this is going to be put off while that gets merged.

After that goes in I'm going to double the thickness values so that defense values are the same as they are now.

**Comment by bombasticSlacks** [comment]
@John-Candlebury I went a bit overboard on integrating the equipment into the Hub quest line and did a bit of future proofing for dialogue for future sci-fi projects. If you want to give the dialogue and interactions a pass (and it passes tests) it's complete.

**Comment by John-Candlebury** [comment]
Was out for the New Year so I didnt get a chance to see the message. 
But I like what you've done with thin anyways.

Theres no Hub01 logo, feel free to draw anything, although a white circle with some black shaded pixels in the center would probably be enough for Ultica's resolution level.

**Comment by bombasticSlacks** [comment]
Well I'm glad! Yeah it ended up being 1 white pixel ahahaha, 

https://github.com/I-am-Erk/CDDA-Tilesets/pull/1149
https://github.com/I-am-Erk/CDDA-Tilesets/pull/1150

I don't know if the envirosuit is intentionally the colors of Urban Camouflage but I took that as inspo

**Comment by andrei8l** [comment]
Some smaller issues:
- the new items don't have any warmth values. Plastic is not breathable and the greaves/vambraces are supposed to be encasing armor pieces so warmth should be quite high
- 10cm length for the greaves is far too low. Knee pads are 11cm and iron greaves are 18cm (which also seems too low). If they're supposed to cover the lower leg from ankle up to above the knee, they should be closer to 40cm. 
Skirt and mantle are also too low, but presumably they fold.
- 0.90L volume for the greaves is only valid if you exclude internal volume, which is not consistent with the older iron/bronze greaves at least
- I can't get the skirts to go in the right pockets even with black/whitelists. Probably not specific to this PR though

And I really question the balance of the mantles and skirts. They are both smaller and lighter than single ESAPI plates, while covering a lot more. They also offer basically the same protection even though they're described as segmented and are apparently flexible enough to fold into a first aid kit.

**Comment by bombasticSlacks** [comment]
Your balance stuff has an incorrect premise which is that any of the volumes or lengths are intentional.

I thought I balanced the volumes and such off similar items but it seems I didn't. These should be light but they are definitely too light. 

I primarily balanced for torso protection which is fine just looking at encumbrance, again volume and weight is a mistake, full set of ESAPI plates is 6 encumbrance for 82% coverage and 25, 50, 50. The soldier mantle is 5 encumbrance for 32, 40, 38 protection. the arms and legs are maybe out of whack by 1-2 but this is supposed to be armor you want to wear and requires a lot of dangerous questing. 

I'll also increase the warmth

Also I can't recreate your issues with the skirt. It should prioritize pockets that it shares a flag with.

**Comment by bombasticSlacks** [comment]
also most items just don't have a longest side. So your greaves example length are low because it would be a massive audit to do them all.

**Comment by AdamChirnside** [comment]
With the strapped layer usage, this conflicts with wearing a Survivor Harness. Shouldn't these be wearable together, or nah?

**Comment by bombasticSlacks** [comment]
It's intentional.  One of the desires for this when I originally talked with John was for it to be a choice as opposed to a molle vest or something similarly "traditional military". 

Fwiw you can still use a single point sling for front carrying a rifle.

**Pull request merged**